maker/maker.go

```diff
@@ -28,59 +28,90 @@ func (m *Method) Lines() []string {
 	return lines
 }
 
-// GetReceiverTypeName returns the name of the
+// GetReceiverTypeName takes in the entire
+// source code and a single declaration.
+// It then checks if the declaration is a
+// function declaration, if it is, it uses
+// the GetReceiverType to check whether
+// the declaration is a method or a function
+// if it is a function we fatally stop.
+// If it is a method we retrieve the type
+// of the receiver based on the types
+// start and end pos in combination with
+// the actual source code.
+// It then returns the name of the
 // receiver type and the function declaration
+//
+// Behavior is undefined for a src []byte that
+// isn't the source of the possible FuncDecl fl
 func GetReceiverTypeName(src []byte, fl interface{}) (string, *ast.FuncDecl) {
 	fd, ok := fl.(*ast.FuncDecl)
 	if !ok {
 		return "", nil
 	}
-	if fd.Recv.NumFields() != 1 {
+	t, err := GetReceiverType(fd)
+	if err != nil {
 		return "", nil
 	}
-	t := fd.Recv.List[0].Type
 	st := string(src[t.Pos()-1 : t.End()-1])
 	if len(st) > 0 && st[0] == '*' {
 		st = st[1:]
 	}
 	return st, fd
 }
 
-// GetParameters ...
-func GetParameters(src []byte, fl *ast.FieldList) ([]string, bool) {
-	if fl == nil {
-		return nil, false
+// GetReceiverType checks if the FuncDecl
+// is a function or a method. If it is a
+// function it returns a nil ast.Expr and
+// a non-nil err. If it is a method it uses
+// a hardcoded 0 index to fetch the receiver
+// because a method can only have 1 receiver.
+// Which can make you wonder why it is a
+// list in the first place, but this type
+// from the `ast` pkg is used in other
+// places than for receivers
+func GetReceiverType(fd *ast.FuncDecl) (ast.Expr, error) {
+	if fd.Recv == nil {
+		return nil, fmt.Errorf("fd is not a method, it is a function")
 	}
-	merged := false
-	parts := []string{}
+	return fd.Recv.List[0].Type, nil
+}
 
+// FormatFieldList takes in the source code
+// as a []byte and a FuncDecl parameters or
+// return values as a FieldList.
+// It then returns a []string with each
+// param or return value as a single string.
+// If the FieldList input is nil, it returns
+// nil
+func FormatFieldList(src []byte, fl *ast.FieldList) []string {
+	if fl == nil {
+		return nil
+	}
+	var parts []string
 	for _, l := range fl.List {
 		names := make([]string, len(l.Names))
-		if len(names) > 1 {
-			merged = true
-		}
 		for i, n := range l.Names {
 			names[i] = n.Name
 		}
-
 		t := string(src[l.Type.Pos()-1 : l.Type.End()-1])
-
-		var v string
+		typeSharingArgs := strings.Join(names, ", ")
```

Comment:
it would make sense to move it into if clause where it really belongs.

Reply:
arh ofc, had moved it outside the if sentence because i actually deleted the if sentence in one iteration, before i realized it was nessecery.